CONTRIBUTING.md

```diff
@@ -1,9 +1,100 @@
+| _if publish=true_ | `cloudflare.products`          | List <=3 products featured in your example                                                    | ["D1", "Durable Objects"]                                                                                              |
+| ❌                | `cloudflare.categories`        | String(s) that map to filter(s) in the template gallery view                                  | ["starter", "storage"]                                                                                                 |
+| _if publish=true_ | `cloudflare.icon_urls`         | Link to icons to make visible on the template card                                            | https://imagedelivery.net/wSMYJvS3Xw-n339CbDyDIA/5ca0ca32-e897-4699-d4c1-6b680512f000/public (default TypeScript logo) |
+| _if publish=true_ | `cloudflare.preview_image_url` | 16:9 aspect screenshot of the template application                                            | (Link will be provided during PR review)                                                                               |
+| ❌                | `cloudflare.publish`           | Boolean to opt-in for a dashboard feature - leave out unless requested by the Cloudflare team | (Primarily for internal contributor use)                                                                               |
+
+### Best Practices: package.json
+
+- **`cloudflare.products`** - Do not exceed 3 product listed, this section of the card component does not overflow nicely. Focus on highlighting the most unique products featured in your template (e.g. majority of our templates leverage 'Worker Assets' in some capacity, but only a select few feature 'DO').
+- **`cloudflare.categories`**
+  - Today, categories are optional to include. In the future, we will support filters in the templates gallery at which point this will become a new template requirement.
+  - Only the following categories are supported: `"starter"`, `"storage"`, and `"ai"`. Anything outside of this set will be rejected by CI.
+- **`cloudflare.preview_image_url`**
+  - Can only be provided by a Cloudflare team member. Image files for icons and preview images are stored in the Cloudflare Templates CF account.
+  - Preview image should be a screenshot of the application running in-browser.
+- **`cloudflare.icon_urls`** - Icon image should be a png of any logos you want to appear on the templates card (most commonly TypeScript, in which case you can use [this URL](https://imagedelivery.net/wSMYJvS3Xw-n339CbDyDIA/5ca0ca32-e897-4699-d4c1-6b680512f000/public)).
+
+### Package-lock.json
+
+All of our templates and Deploy to Cloudflare projects automatically set up Workers CI. In our testing, including a package lock file in the template repository speeds up module resolution by 80% or more.
+
+To generate a package-lock.json for your template, run this command in the root of the repository:
+
+```sh
+pnpm generate-npm-lockfiles
```

Comment:
We suggest `npm run [script]` above. Do we want to standardize on one of pnpm or npm? Also this script name changed
```suggestion
pnpm fix:lockfiles
```